app/models/pipeline_run.rb

```diff
@@ -432,13 +434,7 @@ def monitor_results
 
     # Get pipeline_version, which determines S3 locations of output files.
     # If pipeline version is not present, we cannot load results yet.
-    # [ We use "file_generated_since_run(pipeline_version_file)" because "pipeline_version_file"
-    #   is not in the versioned result folder, so it gets overwritten with each new run.
-    #   TODO: change that, so that we can get rid of "file_generated_since_run".
-    # ]
-    if pipeline_version.blank? && file_generated_since_run(pipeline_version_file)
-      update(pipeline_version: fetch_pipeline_version)
-    end
+    update_pipeline_version(self, :pipeline_version, pipeline_version_file)
```

Comment:
Can you make sure this does "aws s3 cp" only if it has to?   That is, if the version has already been retrieved earlier, it doesn't have to go to s3 for it again.

Reply:
Oh I see that's how it works already.

Reply:
Yes